File: TypeProviders.CSharp.CodeRefactoringProvider.ProviderAttributes/JsonProviderAttribute.cs
using System;

namespace TypeProviders.CSharp
{
    [AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
    public sealed class JsonProviderAttribute : Attribute
    {
        public JsonProviderAttribute(string data)
        {
        }
    }
}

File: TypeProviders.CSharp.BuildTimeGeneration/JsonProviderAttribute.fs
namespace TypeProviders.CSharp.BuildTimeGeneration

open System
open System.Diagnostics
open CodeGeneration.Roslyn

[<AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)>]
[<CodeGenerationAttribute(typeof<JsonProviderGenerator>)>]
[<Conditional("CodeGeneration")>]
type JsonProviderAttribute(data: string) =
    inherit Attribute()
